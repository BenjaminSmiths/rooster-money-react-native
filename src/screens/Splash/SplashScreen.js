import React from 'react'
import { Dimensions, ActivityIndicator } from 'react-native'
import styled from 'styled-components/native'
import roosterLogo from '../../images/roostermoney-logo.png'

const StyledView = styled.View`
    flex-grow: 1;
    justify-content: center;
    align-items: stretch;
    height: ${props => props.height};
`
const LoadingView = styled.View`
    justify-content: center;
    align-items: center;
`
const Logo = styled.Image`
    margin-bottom: 20px;
`

const SplashScreen = () =>
    <StyledView height={Dimensions.get('window').height}>
        <LoadingView>
            <Logo source={roosterLogo}/>
            <ActivityIndicator />
        </LoadingView>
    </StyledView>

export default SplashScreen