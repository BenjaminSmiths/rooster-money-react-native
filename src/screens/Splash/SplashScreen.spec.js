import React from 'react';
import { Provider } from 'mobx-react';
import renderer from 'react-test-renderer';
import SplashScreen from './SplashScreen';


test('Splash Screen Renders Correctly', () => {
    // When
    const tree = renderer.create(
        <Provider>
            <SplashScreen/>
        </Provider>
    ).toJSON();

    // Then
    expect(tree).toMatchSnapshot();
});