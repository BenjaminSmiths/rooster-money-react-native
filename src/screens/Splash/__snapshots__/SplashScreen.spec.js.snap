// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash Screen Renders Correctly 1`] = `
<View
  height={1334}
  style={
    Array [
      Object {
        "alignItems": "stretch",
        "flexGrow": 1,
        "height": 1334,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/roostermoney-logo.png",
        }
      }
      style={
        Array [
          Object {
            "marginBottom": 20,
          },
        ]
      }
    />
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="small"
    />
  </View>
</View>
`;
